tests/test_util.py

```diff
@@ -323,3 +336,98 @@ def flaky_function():
         return "Third time's a charm"
 
     assert flaky_function() == "Third time's a charm"
+
+
+def _merge_index_from_list_serial(index_file_urls: Sequence[Union[str, Tuple[str, str]]],
```

Comment:
similar to what james said on slack, would be great to see benchmarking of serial vs. parallel on more configurations!

Reply:
I have added some more benchmarks in the PR description. PTA~~